VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/services/video-call.service.ts

```diff
@@ -391,6 +403,33 @@ export class VideoCallService {
         this.logger.info(`${this.loggerPrefix} renegotiated`);
     }
 
+    async selectScreenWithMicrophone() {
```

Comment:
Can you move this to its own service and use the onVideoEvidenceShared and onVideoEvidenceStopped in here to manage the streams or even better manage it in user media service use the current stream observable to change in here. This class is already really big.

Reply:
I was debating creating a different class but this implementation is mostly code that already is present. Let's jump on a call to look at what can be done